examples/measure-style.js

```diff
@@ -0,0 +1,266 @@
+  }
+  return output;
+};
+
+const raster = new TileLayer({
+  source: new OSM(),
+});
+
+const source = new VectorSource();
+
+const modify = new Modify({source: source, style: modifyStyle});
+
+function styleFunction(feature, segments, drawType, tip) {
+  const styles = [style];
+  const geometry = feature.getGeometry();
+  const type = geometry.getType();
+  let point, label, line;
+  if (!drawType || drawType === type) {
+    if (type === 'Polygon') {
+      point = geometry.getInteriorPoint();
+      label = formatArea(geometry);
+      line = new LineString(geometry.getCoordinates()[0]);
+    } else if (type === 'LineString') {
+      point = new Point(geometry.getLastCoordinate());
+      label = formatLength(geometry);
+      line = geometry;
+    }
+  }
+  if (segments && line) {
+    let count = 0;
+    line.forEachSegment(function (a, b) {
+      const segment = new LineString([a, b]);
+      const label = formatLength(segment);
+      if (segmentStyles.length - 1 < count) {
+        segmentStyles.push(segmentStyle.clone());
+      }
+      const segmentPoint = new Point(segment.getCoordinateAt(0.5));
+      segmentStyles[count].setGeometry(segmentPoint);
+      segmentStyles[count].getText().setText(label);
+      styles.push(segmentStyles[count]);
+      count++;
+    });
+  }
+  if (label) {
+    labelStyle.setGeometry(point);
+    labelStyle.getText().setText(label);
+    styles.push(labelStyle);
+  }
+  if (
+    tip &&
+    type === 'Point' &&
+    !modify.getOverlay().getSource().getFeatures().length
+  ) {
+    tipStyle.getText().setText(tip);
+    styles.push(tipStyle);
+  }
+  return styles;
+}
+
+const vector = new VectorLayer({
+  source: source,
+  style: function (feature) {
+    return styleFunction(feature, showSegments.checked);
+  },
+});
+
+const map = new Map({
+  layers: [raster, vector],
+  target: 'map',
+  view: new View({
+    center: [-11000000, 4600000],
+    zoom: 15,
+  }),
+});
+
+map.addInteraction(modify);
+
+let draw; // global so we can remove it later
+
+function addInteraction() {
+  const drawType = typeSelect.value;
+  const activeTip =
+    'Click to continue drawing the ' +
+    (drawType === 'Polygon' ? 'polygon' : 'line');
+  const idleTip = 'Click to start measuring';
+  let tip = idleTip;
+  draw = new Draw({
+    source: source,
+    type: drawType,
+    style: function (feature) {
+      return styleFunction(feature, showSegments.checked, drawType, tip);
+    },
+  });
+  draw.on('drawstart', function () {
+    if (clearPrevious.checked) {
+      source.clear();
+    }
+    modify.setActive(false);
+    tip = activeTip;
+  });
+  draw.on('drawend', function () {
+    modify.setActive(true);
+    tip = idleTip;
+  });
+  modify.setActive(true);
+  map.addInteraction(draw);
+}
+
+typeSelect.onchange = function () {
+  map.removeInteraction(draw);
+  addInteraction();
+};
+
+addInteraction();
+
+showSegments.onchange = function () {
+  vector.changed();
```

Comment:
If you are currently drawing the labels of that feature are not immediatly updated.
```suggestion
  vector.changed();
  draw.getOverlay().changed();
```

Reply:
The draw labels are now updating immediately, and by temporarily overriding the modify style geometry I was able to immediately display the tip in the expected position.